sdk/cosmosdb/cosmos/test/public/integration/changeFeedIterator.spec.ts

```diff
@@ -435,3 +442,461 @@ describe("test changefeed for feed range", function () {
     assert.equal(counter, 40, "40 items should be fetched");
   });
 });
+
+describe("test changefeed allVersionsAndDeletes mode for entire container", function () {
+  let container: Container;
+  before(async function () {
+    const newTimeStamp = ChangeFeedRetentionTimeSpan.fromMinutes(5);
+    const changeFeedPolicy = new ChangeFeedPolicy(newTimeStamp);
+    const containerDef: ContainerDefinition = {
+      partitionKey: {
+        paths: ["/name"],
+        version: PartitionKeyDefinitionVersion.V1,
+      },
+      changeFeedPolicy: changeFeedPolicy,
+    };
+    const throughput: RequestOptions = { offerThroughput: 21000 };
+    container = await getTestContainer(
+      "Changefeed allVersionsAndDeletes-EntireContainer",
+      undefined,
+      containerDef,
+      throughput
+    );
+    for (let i = 1; i < 6; i++) {
+      await container.items.create({ id: `sample1+${i}`, name: "sample1", key: i });
+      await container.items.create({ id: `sample1+${i}`, name: "sample2", key: i });
+      await container.items.create({ id: `sample1+${i}`, name: "sample3", key: i });
+      await container.items.create({ id: `sample1+${i}`, name: "sample4", key: i });
+    }
+  });
+  it("startFromBeginning is not supported", async function () {
+    try {
+      const changeFeedIteratorOptions: ChangeFeedIteratorOptions = {
+        changeFeedStartFrom: ChangeFeedStartFrom.Beginning(),
+        changeFeedMode: ChangeFeedMode.AllVersionsAndDeletes,
+      };
+      const iterator = container.items.getChangeFeedIterator(changeFeedIteratorOptions);
+
+      while (iterator.hasMoreResults) {
+        await iterator.readNext();
+      }
+    } catch (err: any) {
+      assert.strictEqual(err.code, 400);
```

Comment:
A status code of 400 may be returned for various other reasons. Can we implement more stringent checks?